six/modules/c++/cphd/include/cphd/MetadataBase.h

```diff
@@ -0,0 +1,87 @@
+/* =========================================================================
+ * This file is part of cphd-c++
+ * =========================================================================
+ *
+ * (C) Copyright 2004 - 2019, MDA Information Systems LLC
+ *
+ * cphd-c++ is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; If not,
+ * see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef __CPHD_METADATA_BASE_H__
+#define __CPHD_METADATA_BASE_H__
+
+#include <ostream>
+#include <six/Init.h>
+
+namespace cphd
+{
+/*!
+ *  \class MetadataBase
+ *  \brief Abstract base class for the CPHD and CPHD03 Metdata objects.
+ *
+ *  The metadata object is derived for CPHD and CPHD03
+ *. This class provides the interface needed
+ *  to interact with the signal block reader Wideband, currently in CPHD
+ *
+ */
+struct MetadataBase
```

Comment:
This name isn't ideal, but I can try to think of something if you think it needs changing